### PR TITLE
feat(db): gym kiosk preset schema + @boardsesh/kiosk config package

### PR DESCRIPTION
**Pull request by marcodejongh**
## Summary

PR 1 of the gym-kiosk-dashboard epic: the database foundation plus the shared kiosk-config package. No backend resolvers and no web UI — those land in later PRs built on this schema.

**Pivot (before merge):** the kiosk is now an **opinionated preset surface**, not a free-form dashboard. The first draft of `@boardsesh/kiosk` modelled a free-form 12×8 widget grid; this PR replaces it with the preset config below so the dead grid schema never reaches `main`. V1 targets **1–4 boards per kiosk** (single/dual/triple/quad — the preset is derived from how many boards are configured) plus an **optional single leaderboard rail**. No drag-and-drop, no grid coordinates, no widget types.

**What's here**

- **`gym_kiosks` table** (`packages/db/src/schema/app/gym-kiosks.ts`) — one row per smart-TV dashboard a gym runs. `bigserial` PK, unique `uuid`, `gym_id` FK → `gyms` (cascade delete), `slug`, `name`, a `layout` jsonb typed `$type<KioskLayout>` (type-only import from `@boardsesh/kiosk`) defaulting to `{"version":1,"boards":[],"leaderboard":null}`, and created/updated/deleted timestamps. Partial unique index on `(gym_id, slug) WHERE deleted_at IS NULL` (one live kiosk per slug, freed on soft-delete) plus a partial `gym_id` index.
- **4 branding columns on `gyms`** — `logo_url`, `brand_primary_color`, `brand_accent_color`, `brand_background_color` (all nullable text). `logo_url` is the transparent brand mark, kept distinct from `image_url` (the gym photo); the colours are `#RRGGBB` hex (zod-validated in PR 2). Comments in the schema document both distinctions.
- **`@boardsesh/kiosk`** (`packages/shared/kiosk/`) — pure TS + zod, the single source of truth for kiosk layout config shared by the backend write-validator and the web renderer/editor.
  - `constants.ts` — `KIOSK_LAYOUT_VERSION` (stays `1`: the grid draft never shipped, so version 1 has only ever meant "preset config"), `MAX_KIOSK_BOARDS = 4`, `MAX_KIOSKS_PER_GYM = 10`, `KIOSK_PRESETS = ['single','dual','triple','quad']` + `kioskPresetForBoardCount(count)` (returns `null` for 0/out-of-range/non-integer → the TV shows a "not set up yet" placeholder), `KIOSK_LEADERBOARD_PERIODS = ['session','day','week','month']` (default `'session'`), `KIOSK_SESSION_WINDOW_MINUTES = 180`.
  - `kiosk-config.ts` (renamed from `widget-config.ts`) — `KioskBoardSlotSchema` (`{ boardUuid }`, an object so additive per-slot options don't need a version bump; slot order = on-screen order, no slot id), `KioskLeaderboardSchema` (`{ boardUuid: uuid|null, period }`, `null` scopes the rail to all kiosk boards), and `KioskLayoutSchema` (`{ version: 1, boards: slot[] ≤4, leaderboard: nullable }`). A `superRefine` adds the two cross-field rules the object schema can't: no duplicate `boardUuid`, and a single-board rail must point at a board the kiosk actually shows. `boards: []` is valid on purpose (create-then-configure). `parseKioskLayoutLenient` is the forward-compatible reader: unknown version / non-object → empty layout; per-slot `safeParse` dropping invalid slots, dedupe by `boardUuid` keeping the first, cap at 4; a malformed rail → `null`; a single-board rail whose board didn't survive the slot repair is **widened** to all-boards rather than dropped. It returns `{ layout, droppedBoardCount, droppedLeaderboard }`.
  - 34 unit tests (`kiosk-config.test.ts`): strict parse per shape, both `superRefine`s, unknown-key stripping at every nesting level (the epic's additive-field forward-compat contract), lenient drops/dedupe/cap/version-cutover/rail-widening (including a rail orphaned by the cap), empty default, and `kioskPresetForBoardCount` bounds.
- **Migration `0177_nervous_machine_man`** — regenerated by `drizzle-kit generate` after the schema-default change. Test-DDL mirrored into `packages/backend/src/__tests__/schema-sql.ts` (same preset default).

## Migration footgun note

The repo documents a footgun (`packages/db/src/schema/app/ascents.ts:180-188`): `drizzle-kit generate` historically emits a spurious destructive `ALTER TABLE "boardsesh_ticks" DROP CONSTRAINT "boardsesh_ticks_quality_range"` and drops the `checkConstraints` block from the new snapshot. **This did not reproduce with the pinned `drizzle-kit@0.31.10`** on regeneration. The regenerated `0177` SQL contains only the intended statements (no `DROP CONSTRAINT`), and `0177_snapshot.json` preserves the `boardsesh_ticks_quality_range` check block. Both verified explicitly, so no manual SQL/snapshot surgery was needed. The final migration SQL is exactly: the `gym_kiosks` CREATE TABLE + FK + 2 indexes, plus the 4 `gyms` ADD COLUMN statements.

## Non-standard choices (flagged)

- **`@boardsesh/kiosk` is a `packages/db` dependency**, imported type-only in the drizzle file. This mirrors the existing `sessions.ts` pattern (`import type { ClimbQueueItem } from '@boardsesh/shared-schema'` feeding `queue.$type<ClimbQueueItem[]>()`); `packages/db` already declares workspace deps. The type import is erased at compile time — the compiled drizzle output has no runtime dependency on the package (verified). The `layout` default is a hardcoded literal (not `KIOSK_LAYOUT_VERSION`) precisely so no value import leaks into the generated drizzle output; a version bump is a documented hard cutover that must update the literal in lockstep.
- **A `typecheck:kiosk` task is registered** in `vite.config.ts` — since `@boardsesh/kiosk` carries real zod logic, it's in both the vitest projects list and the typecheck aggregate so `vp run typecheck` validates it fully.
- **`mapRawGymRow` in `resolvers/social/gyms.ts`** carries the 4 new branding fields. This is a required compile-fix from adding the columns (its return type is `gyms.$inferSelect`), not a feature change — the raw PostGIS proximity query already uses `SELECT *`, so the columns are present in the row and mapped through. Branding is not yet surfaced in the SDL (that's PR 2).

## Test plan

- `vp check` on the changed files — clean (0 warnings/lint/type errors across all 6 changed files). CI lints changed files on PRs; the full-repo warning baseline is pre-existing and untouched by this diff.
- `vp run typecheck` (full) — **pass** (web `next build` + `tsc` and every package green).
- `vp test run --project kiosk` — **34/34 pass**.
- **Migration applied to real Postgres 17** — the regenerated `0177` SQL ran against an isolated throwaway DB (with a `gyms` stub) under `ON_ERROR_STOP=1`: every statement succeeded; the 4 gyms columns, all `gym_kiosks` indexes/FK, and the jsonb default were verified present; a test insert materialised `layout` as `{"version":1,"boards":[],"leaderboard":null}` (`boards` is a JSON array, `leaderboard` is JSON null). The DB was then dropped. (A full from-scratch `db:migrate` chain isn't runnable here — the dev DB is a pre-built image whose board-catalog tables come from data imports, not migrations, so an unrelated early migration fails on an empty DB.)
- Backend test suite not run full here (concurrent epic sessions share the same postgres/redis and collide, per the documented `VITEST_POOL_ID` isolation gap); the fresh-DB migration proof above covers the schema change, and CI builds isolated worker DBs from this branch's `schema-sql.ts`.

## Release Notes

- [x] No release note needed (internal / technical change)

### CODE REVIEW

File: packages/shared/kiosk/src/widget-config.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    .int()
+    .min(0)
+    .max(KIOSK_GRID_COLUMNS - 1),
+  y: z
+    .number()
+    .int()
+    .min(0)
+    .max(KIOSK_GRID_ROWS - 1),
+  w: z.number().int().min(1).max(KIOSK_GRID_COLUMNS),
+  h: z.number().int().min(1).max(KIOSK_GRID_ROWS),
```
[the remark this answers is not in the document]
Comment (marcodejongh) [reply packages/shared/kiosk/src/widget-config.ts]: Declined as superseded: this refers to the widget-grid draft (`GridPositionSchema` / `widget-config.ts`), which never merged — the package was rewritten to the preset schema (1-4 board slots + optional leaderboard rail, no grid coordinates) in 7bafac9a3 before merge. There is no span/position field left to overflow.

**Comment by marcodejongh** [comment]
Review replies (the schema-v2 rewrite in 7bafac9a3 replaced the widget-grid draft pre-merge, so the three reviews span two different schemas — responding per item):

**Review of 2026-07-15 07:21 (pre-pivot, widget grid):**
- Grid overflow (`x + w` / `y + h`): superseded — `GridPositionSchema` and the whole grid model were deleted in 7bafac9a3. The preset schema has no positions or spans.
- Hardcoded version literal in the drizzle default: unchanged by design; the schema comment documents the lockstep rule and why a value import would break `drizzle-kit generate`.
- Slug format/length at DB level: declined — PR 3 (#3633) ships `KioskSlugSchema` as the write gate for format and length; the column deliberately stays plain text so a soft-deleted row can free an oddly-shaped legacy slug (documented in the schema comment).

**Review of 2026-07-15 10:19:**
1. `logo_url` URL validation at the DB layer: declined — validation lives in the backend zod write layer (#3631's `GymLogoUrlSchema`, tightened to a full pattern). No DB text column self-validates in this repo; the DB stores what the validated write layer hands it.
2. Leaderboard `boardUuid` vs real board records: by design out of scope for the shared package (pure TS, no DB access). PR 3 (#3633) write validation checks alive-and-gym-linked over both `layout.boards` and `leaderboard.boardUuid`.
3. Missing trailing newline in `0177_nervous_machine_man.sql`: fixed in 64daff482.
4. Missing `{ version: 1 }` keys-absent edge-case test: added in 64daff482.
5. `MAX_KIOSKS_PER_GYM` DB enforcement: declined at DB level — a per-gym row-count cap needs a trigger, not a CHECK, and the repo enforces such invariants at the write layer. PR 3 (#3633) enforces it in `createGymKiosk`.

**Review of 2026-07-15 10:45:**
1. Lenient reader reporting `droppedBoardCount: 0` on a version mismatch: real — fixed in 64daff482. An unrecognised version still returns the empty layout, but the flags now report the stored board count and rail presence so callers surfacing data-loss warnings see the wholesale discard. Non-object garbage keeps zero flags (nothing interpretable was stored). The version-cutover test now pins the new contract.
2. Slug format at DB: see above — PR 3's `KioskSlugSchema` owns it.
3. `updatedAt` not auto-updated: matches repo convention as noted; PR 3's update resolver sets it explicitly.
4. Trailing newline: fixed in 64daff482.
